sdk/core/core-rest-pipeline/src/policies/formDataPolicy.browser.ts

```diff
@@ -17,25 +18,30 @@ export function formDataPolicy(): PipelinePolicy {
     name: formDataPolicyName,
     async sendRequest(request: PipelineRequest, next: SendRequest): Promise<PipelineResponse> {
```

Comment:
I think I found a much easier way to handle the encoding that avoids needing us to have a special helper:

```ts
async sendRequest(request: PipelineRequest, next: SendRequest): Promise<PipelineResponse> {
  if (request.formData) {
    const formData = request.formData;
    const requestForm = new FormData();
    for (const formKey of Object.keys(formData)) {
      const formValue = formData[formKey];
      if (Array.isArray(formValue)) {
        for (const subValue of formValue) {
          requestForm.append(formKey, subValue);
        }
      } else {
        requestForm.append(formKey, formValue);
      }
    }

    request.body = requestForm;
    request.formData = undefined;
    const contentType = request.headers.get("Content-Type");
    if (contentType && contentType.indexOf("application/x-www-form-urlencoded") !== -1) {
      request.body = new URLSearchParams(requestForm).toString();
    } else if (contentType && contentType.indexOf("multipart/form-data") !== -1) {
      // browser will automatically apply a suitable content-type header
      request.headers.delete("Content-Type");
    }
   
  }
  return next(request);
}
```

Reply:
This looks great! I didn't know that we can construct URLSearchParams with FormData.